deployment/custom-deployment/bin/update-es-logs-and-client-stack-vars.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+source .env
+
+#STACK
+
+echo "==update %%CLIENT_APP_BUCKET%% in stack with $2=="
+replace="s/%%CLIENT_APP_BUCKET%%/$ClientAppBucketName/g"
+sed -i -e $replace ./lib/cdk-textract-client-stack.js
+
+echo "==update Elastic Search Cluster ($ElasticSearchCluster) with log streams to Log Groups: $ElasticSearchSearchLogGroup and $ElasticSearchIndexLogGroup"
```

Comment:
Is there a specific reason to the script? 
I'd like to reduce the functionality in scripts where if we do get an error, it is difficult to re-run the steps.
Could we use a custom resource created as a part of the Client stack to add this in?
Custom resource snippet available here : https://github.com/aws/aws-cdk/issues/5343

Reply:
This is not a new script, just an expanded "stack-variables-to-client" script.

While I like the idea in principal of using a custom resource, in this case I actually think it makes it more complex, and harder to debug. When custom resources fail, they often do so silently, and you have to dig into lambda logs to see the error. AWS CLI errors from this script are very transparent, showing up in stdout for both the CICD and Dev deploy, and you can re-run just this part of the deployment by rerunning only the script. Redeploying the stack to retest the custom resource would take longer.

The custom resource also requires quite a bit more added code, e.g another lambda directory and .js file. We also would still need to add functionality to this script because we would need to substitute the ES domain name into the Client CDK stack so that it knows what to target. With this method, its just a few added lines in this file, and a few added lines in pre-build following an established pattern.

Reply:
Fair point.